### DISCUSSION TITLE
MOU - risk as an objective and uncertainty propagation

### DISCUSSION THREAD

**Discussion by cnicol-gwlogic** [usgs/pestpp#237] [Q&A] [answered]
Hi again,

When running MOU ([5.2.0 beta](https://github.com/usgs/pestpp/releases/tag/5.2.0.b)) with risk included (as an objective in my case), I am seeing negative values for some objectives when that is not actually possible in the raw model outputs that are read by pest via instruction files. Some of these get quite large, and seem to throw my optimisation out of whack bigtime compared with what I was seeing in a case with specified risk of 0.5 (i.e., just optimising dv_pars on a single base case model parameter realisation; noting that this was using the previous release exe). When I look in the current obs.pop.csv, I see none of these negative values - they all make sense. Note that I am using opt_chance_points(single) - so I wonder if it is arising from the process of propagating uncertainty from that single dv_par realisation to all the others (99). 

Is this actually possible, or do you reckon I've stuffed something up?  If this is a consequence of chance_points(single), I wonder if a nice option might be opt_chance_points(n) - n aimed at using representative selection of spread in dv_par members.

Thanks a lot.

**Comment by jtwhite79** [comment, marked as answer]
Its possible that the chance reuse combined with risk-shifting can produce non-physical values...Can you post your rec file so we can have a look?

**Reply by cnicol-gwlogic** [reply]
No worries, appreciate it, again. 
[dubv1tr02_mou1v06.rec.zip](https://github.com/usgs/pestpp/files/10680559/dubv1tr02_mou1v06.rec.zip)
Example objective to look at is USECOL:VALUE_AVERAGE_ANNUAL_VOLUME - should not go negative. One of the others is also affected (USECOL:DIFF_FROM_BASELINE_COUNT_BORES_IMPACTED_BY_AT_LEAST_1_YEARS), although that can go slightly negative, just nowhere near -409 or whatever the min estimate is.

**Reply by cnicol-gwlogic** [reply]
...starting to think i've messed something up, at least with USECOL:DIFF_FROM_BASELINE_COUNT_BORES_IMPACTED_BY_AT_LEAST_1_YEARS. Initial population pre-shift minimum value looks bananas to me. Previous runs had a value of 1, not -415. I did re-order a few things in the batch file between runs, so don't waste too much of your time on this, thanks J.

**Reply by cnicol-gwlogic** [reply]
Definitely an issue at my end. Sorry for the unnecessary hassle.

**Reply by jtwhite79** [reply]
all good @cnicol-gwlogic
